NitroxClient/Communication/PacketSuppression.cs

```diff
@@ -1,21 +1,22 @@
 ﻿using System;
-using System.Collections.Generic;
+using NitroxModel.Packets;
 
-namespace NitroxClient.Communication
-{
-    public class PacketSuppressor<T> : IDisposable
-    {
-        private readonly HashSet<Type> suppressedPacketTypes;
+namespace NitroxClient.Communication;
 
-        public PacketSuppressor(HashSet<Type> suppressedPacketTypes)
-        {
-            this.suppressedPacketTypes = suppressedPacketTypes;
-            suppressedPacketTypes.Add(typeof(T));
-        }
+/// <summary>
+///     Suppresses the given packet type from being sent. Disables the suppression when disposed.
+/// </summary>
+/// <typeparam name="T">The packet type to suppress.</typeparam>
+public readonly struct PacketSuppressor<T> : IDisposable
+    where T : Packet
+{
+    private static bool isSuppressed;
+    public static bool IsSuppressed => isSuppressed;
```

Comment:
Do you think we would get better performance if we stuck with the previous method but used a static hashmap (that does NOT vary with T)?  We would have to hack it to allow multiple disposes (or the api may already support but not sure).  This would reduce object creation in favor of hashmap lookup.